crates/python/src/execution_data.rs

```diff
@@ -75,6 +73,14 @@ py_wrap_type! {
 
 #[pymethods]
 impl PyRegisterMatrix {
+    fn to_ndarray(&self, py: Python<'_>) -> Py<PyAny> {
+        self.as_integer(py)
+            .map(|array| array.into_py(py))
+            .or(self.as_real(py).map(|array| array.into_py(py)))
+            .or(self.as_complex(py).map(|array| array.into_py(py)))
+            .expect("A RegisterMatrix can't be any other type.")
+    }
```

Comment:
Is there a way to write this that does not involve possible panics?

Reply:
We could return a `PyResult` to raise a different exception. pyo3 already raises a [PanicException](https://pyo3.rs/main/doc/pyo3/panic/struct.panicexception) when Rust code called from Python panic. It will usually unwind the whole stack, which I think is what we want if this block somehow fails the expectation.